src/App.tsx

```diff
@@ -14,11 +17,104 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface Type {
+  [key:string]: string,
+}
+
+  type State = {
+    selectGoods: string[],
+    buttonCheck: boolean,
+  };
+
+class App extends React.PureComponent<{}, State> {
+  goodsStorage:Type = { Jam: 'Jam' };
```

Comment:
why did you decide to use an Object to store data? your key-value pairs have the same keys and values. i think an array is more appropriate here.